src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -5500,16 +5502,28 @@ public object VisitArrayExpression(ArrayExpressionAst arrayExpressionAst)
             }
             values = values ?? CaptureAstResults(subExpr, CaptureAstContext.Enumerable);
 
-            if (values.Type.IsArray)
+            if (values.Type == typeof(object[]) || values.Type == typeof(List<object>))
```

Comment:
The goal of the optimization was to cover `@(1,2)` - so maybe the original test was too general. This fix ends up adding unnecessary overhead to the optimization by cloning and adding an `if` test.

I think a better fix is to just constrain the optimization more - basically `if (values is ArrayLiteralAst)`.

I don't recall why I added the `List<object>` case.